### PR TITLE
CLOUDSTACK-10190:Duplicate public VLAN for two different admin accounts.

### PR DESCRIPTION
**Pull request by niteshsarda**
**ISSUE :**
Duplicate public VLAN for two different admin accounts.

**STEPS TO REPRODUCE :**
1. Start multiple threads for executing createVlanIpRange API.
2. Make sure multiple threads run in parallel.
3. Verify vlan table in DB, duplicate entry for same VLAN and IP address range will be encountered, just id and uuid will be different, rest all fields will have similar value.
4. Following entry will be observed in vlan table :
`mysql> select * from vlan where vlan_id like 'vlan://77' and removed is null;
+----+--------------------------------------+-----------+--------------+-----------------+---------------------------+----------------+----------------+------------+---------------------+-------------+----------+-----------+---------+---------------------+
| id | uuid                                 | vlan_id   | vlan_gateway | vlan_netmask    | description               | vlan_type      | data_center_id | network_id | physical_network_id | ip6_gateway | ip6_cidr | ip6_range | removed | created             |
+----+--------------------------------------+-----------+--------------+-----------------+---------------------------+----------------+----------------+------------+---------------------+-------------+----------+-----------+---------+---------------------+
| 15 | 6a205b78-d162-43e3-8da9-86a3ff60f40e | vlan://77 | 10.112.63.65 | 255.255.255.192 | 10.112.63.66-10.112.63.70 | VirtualNetwork |              1 |        200 |                 200 | NULL        | NULL     | NULL      | NULL    | 2017-12-13 12:55:51 |

| 17 | ff8b5175-b247-45a5-b8d3-feb6a1ca64d0 | vlan://77 | 10.112.63.65 | 255.255.255.192 | 10.112.63.66-10.112.63.70 | VirtualNetwork |              1 |        200 |                 200 | NULL        | NULL     | NULL      | NULL    | 2017-12-13 12:55:51 |
+----+--------------------------------------+-----------+--------------+-----------------+---------------------------+----------------+----------------+------------+---------------------+-------------+----------+-----------+---------+---------------------+
`

### CODE REVIEW

File: server/src/com/cloud/configuration/ConfigurationManagerImpl.java [hunk 1]
```diff
@@ -2859,35 +2859,42 @@ public Vlan createVlanAndPublicIpRange(final CreateVlanIpRangeCmd cmd) throws In
...
             final String ip6Gateway, final String ip6Cidr, final Domain domain, final Account vlanOwner, final Network network, final Pair<Boolean, Pair<String, String>> sameSubnet) {
-        return Transaction.execute(new TransactionCallback<Vlan>() {
-            @Override
-            public Vlan doInTransaction(final TransactionStatus status) {
-                String newVlanNetmask = newVlanNetmaskFinal;
-                String newVlanGateway = newVlanGatewayFinal;
+        final GlobalLock commitVlanLock = GlobalLock.getInternLock("CommitVlan");
+        commitVlanLock.lock(5);
+        s_logger.debug("Acquiring lock for committing vlan");
+        try {
```
Comment (niteshsarda) [inline server/src/com/cloud/configuration/ConfigurationManagerImpl.java]: Below is an existing code, changes can be seen due to addition of spaces for doing proper alignment.

**Comment by rhtyd** [comment]
@blueorangutan package

**Comment by rhtyd** [comment]
@blueorangutan test

**Comment by niteshsarda** [comment]
Test Cases which are failing are not related to these changes.

**Comment by rhtyd** [comment]
Test LGTM, ignoring known failures. Additional review requested.

**Comment by mrunalinikankariya** [comment]
Code LGTM

**Comment by niteshsarda** [comment]
@rhtyd : Can you please merge this PR as code LGTM is given by @mrunalinikankariya ?

**Comment by rhtyd** [comment]
Merging based on code review and test results.

**Pull request merged**